### DISCUSSION TITLE
Access Result and Update methods in ApolloClient Query

### DISCUSSION THREAD

**Discussion by fernandoagarcia** [Akryum/vue-apollo#550] [Q&A]
Is it possible to access the 'result' and 'update' methods using the query function from apolloclient similar to the way in the DollarApollo api? I am using this via Vuex with the following code: 

`  async fetchEventDetails ({commit}, d) {

    try {
      const response = await apolloClient.query({
        query: EVENT_DETAILS,
        variables: { id: d },
        result: (ApolloQueryResult, key) => {
          // Nothing happens here
          console.log(ApolloQueryResult)
          console.log(key)
        }
      })

      const res = response.data[EVENT_TYPE.QUERY.DETAILS]

      const data = {
        id: res.id,
        label: res.name,
        campaign: null,
        description: isStringWithContent(res.description)
          ? res.description
          : null,
        startDate: res.eventStartDt,
        endDate: null,
        type: res.campaignEventType,
        venue: {
          label: null,
          address1: null,
          address2: null,
          city: null,
          state: null,
          zip: null
        }
      }

      commit('SET_EVENT', {...event, ...data})
    } catch (err) {
      mergeNotifications(err.graphQLErrors.map(({message}, i) => ({
        icon: 'fal fa-ban',
        msg: message,
        id: `graphQLErrors-${i}`,
        status: 'failed'
      })))
      throw err
    }
  },`

I would like to be able to access more of the DollarApollo api in Vuex if at all possible.

**Comment by ydfzgyj** [comment]
`DollarApollo` will create a smart query, and query function will only create a normal apollo query.
You can use `this.$apollo.addSmartQuery` to create a smart query manually.
